indico/modules/events/editing/controllers/backend.py

```diff
@@ -7,32 +7,193 @@
+        'preserved_args': {'type'}
+    }
+
+    def _process_args(self):
+        RHContributionDisplayBase._process_args(self)
+        self.editable = (Editable.query
+                         .with_parent(self.contrib)
+                         .filter_by(type=EditableType[request.view_args['type']])
+                         .first())
+
+    def _check_access(self):
+        RHContributionDisplayBase._check_access(self)
+        if not session.user:
+            raise Forbidden
+
+    def _user_is_authorized_submitter(self):
+        if session.user.is_admin:
+            # XXX: not sure if we want to keep this, but for now it's useful to have!
+            return True
+        return self.contrib.is_user_associated(session.user, check_abstract=True)
+
+    def _user_is_authorized_editor(self):
+        if session.user.is_admin:
+            # XXX: not sure if we want to keep this, but for now it's useful to have!
+            return True
+        return self.editable.editor == session.user
+
+
+class RHContributionEditableRevisionBase(RHContributionEditableBase):
+    """Base class for operations on the latest revision of an Editable."""
+
+    normalize_url_spec = {
+        'locators': {
+            lambda self: self.contrib
+        },
+        'preserved_args': {'type', 'revision_id'}
+    }
+
+    def _process_args(self):
+        RHContributionEditableBase._process_args(self)
+        if not self.editable:
+            raise NotFound
+        self.revision = self.editable.revisions[-1]
+        if self.revision.id != request.view_args['revision_id']:
+            raise UserValueError(_('Only the latest revision can be updated'))
+
+    def _check_revision_access(self):
+        raise NotImplementedError
+
+    def _check_access(self):
+        RHContributionEditableBase._check_access(self)
+        if not self._check_revision_access():
+            raise UserValueError(_('You cannot perform this action on this revision'))
+
+
+class RHEditable(RHContributionEditableBase):
+    """Retrieve an Editable with all its data."""
+
+    def _check_access(self):
+        RHContributionEditableBase._check_access(self)
+        if self.event.can_manage(session.user):
+            return
+        if not self._user_is_authorized_editor() and not self._user_is_authorized_editor():
+            raise Forbidden
 
     def _process(self):
         return EditableSchema().jsonify(self.editable)
+
+
+class RHCreateEditable(RHContributionEditableBase):
+    """Create a new Editable for a contribution."""
+
+    def _check_access(self):
+        RHContributionEditableBase._check_access(self)
+        if not self._user_is_authorized_submitter():
+            # XXX: should event managers be able to submit on behalf of the user?
+            raise Forbidden
+        # TODO: check if submitting papers for editing is allowed in the event
+
+    def _process(self):
+        if self.editable:
+            raise UserValueError('Editable already exists')
```

Comment:
Missing i18n?

Reply:
If an error is not going to show up to users during normal usage there's not much benefit in translating it. But in theory this one could happen if you have multiple tabs open and already submitted your editable in another one, so can't hurt to add i18n to it..